Incremental OCaml builds don't work

This is quite annoying. If you run this command

```
ARCH=RV64 make ocaml_emulator/riscv_ocaml_sim_RV64
```

and you have some error in your OCaml code it will do the expected thing, compile the Sail, print your OCaml error. If you immediately run it *again* you'll get this error:

```
mkdir -p ocaml_emulator/_sbuild
cp ocaml_emulator/_tags ocaml_emulator/platform.ml ocaml_emulator/platform_impl.ml ocaml_emulator/softfloat.ml ocaml_emulator/riscv_ocaml_sim.ml generated_definitions/ocaml/RV64/*.ml ocaml_emulator/_sbuild
cp: cannot create regular file 'ocaml_emulator/_sbuild/elf_loader.ml': Permission denied
cp: cannot create regular file 'ocaml_emulator/_sbuild/sail_lib.ml': Permission denied
cp: cannot create regular file 'ocaml_emulator/_sbuild/util.ml': Permission denied
make: *** [Makefile:222: ocaml_emulator/_sbuild/riscv_ocaml_sim.native] Error 1
```

Looking at those files for some reason they are made read-only:

```
❯ ls -l ocaml_emulator/_sbuild/
total 4776
drwxr-xr-x 2 timothy.hutt domain users    4096 Feb 12 15:11 _build
-r--r--r-- 1 timothy.hutt domain users   11857 Feb 12 15:11 elf_loader.ml
-rw-r--r-- 1 timothy.hutt domain users    6753 Feb 12 15:20 platform_impl.ml
-rw-r--r-- 1 timothy.hutt domain users    5874 Feb 12 15:20 platform.ml
-rw-r--r-- 1 timothy.hutt domain users 4771832 Feb 12 15:20 riscv.ml
-rw-r--r-- 1 timothy.hutt domain users    7339 Feb 12 15:20 riscv_ocaml_sim.ml
-r--r--r-- 1 timothy.hutt domain users   41396 Feb 12 15:11 sail_lib.ml
-rw-r--r-- 1 timothy.hutt domain users    1914 Feb 12 15:20 softfloat.ml
-rw-r--r-- 1 timothy.hutt domain users     163 Feb 12 15:20 _tags
-r--r--r-- 1 timothy.hutt domain users   22051 Feb 12 15:11 util.ml
```

They look like they are from the Sail compiler so I guess it has something to do with how I installed that, which is just via `make install`. From the Sail source directory:

```
❯ ls -l `find . -name 'sail_lib.ml'`
-r--r--r-- 1 timothy.hutt domain users 41396 Dec 12 15:59 ./_build/default/src/lib/sail_lib.ml
lrwxrwxrwx 1 timothy.hutt domain users    39 Dec 12 15:59 ./_build/install/default/lib/libsail/sail_lib.ml -> ../../../../default/src/lib/sail_lib.ml
lrwxrwxrwx 1 timothy.hutt domain users    45 Dec 12 15:59 ./_build/install/default/share/sail/src/lib/sail_lib.ml -> ../../../../../../default/src/lib/sail_lib.ml
-rw-r--r-- 1 timothy.hutt domain users 41396 Feb  9 15:58 ./src/lib/sail_lib.ml
```

This seems to be something that Dune [deliberately does](https://github.com/ocaml/dune/issues/5570) so I have no idea what's going on here or how it was supposed to work.

--- a/Makefile
+++ b/Makefile
@@ -226,14 +226,18 @@ generated_definitions/ocaml/$(ARCH)/riscv.ml: $(SAIL_SRCS) Makefile
 	mkdir -p generated_definitions/ocaml/$(ARCH)
 	$(SAIL) $(SAIL_FLAGS) -ocaml -ocaml-nobuild -ocaml_build_dir generated_definitions/ocaml/$(ARCH) -o riscv $(SAIL_SRCS)
 
+# cp -f is required because the generated_definitions/ocaml/$ARCH/*.ml files can
+# be read-only, which would otherwise make subsequent builds fail.
 ocaml_emulator/_sbuild/riscv_ocaml_sim.native: generated_definitions/ocaml/$(ARCH)/riscv.ml ocaml_emulator/_tags $(PLATFORM_OCAML_SRCS) Makefile
 	mkdir -p ocaml_emulator/_sbuild
-	cp ocaml_emulator/_tags $(PLATFORM_OCAML_SRCS) generated_definitions/ocaml/$(ARCH)/*.ml ocaml_emulator/_sbuild
+	cp ocaml_emulator/_tags $(PLATFORM_OCAML_SRCS) ocaml_emulator/_sbuild
+	cp -f generated_definitions/ocaml/$(ARCH)/*.ml ocaml_emulator/_sbuild
 	cd ocaml_emulator/_sbuild && ocamlbuild -use-ocamlfind riscv_ocaml_sim.native
 
 ocaml_emulator/_sbuild/coverage.native: generated_definitions/ocaml/$(ARCH)/riscv.ml ocaml_emulator/_tags.bisect $(PLATFORM_OCAML_SRCS) Makefile
 	mkdir -p ocaml_emulator/_sbuild
-	cp $(PLATFORM_OCAML_SRCS) generated_definitions/ocaml/$(ARCH)/*.ml ocaml_emulator/_sbuild
+	cp $(PLATFORM_OCAML_SRCS) ocaml_emulator/_sbuild
+	cp -f generated_definitions/ocaml/$(ARCH)/*.ml ocaml_emulator/_sbuild
 	cp ocaml_emulator/_tags.bisect ocaml_emulator/_sbuild/_tags
 	cd ocaml_emulator/_sbuild && ocamlbuild -use-ocamlfind riscv_ocaml_sim.native && cp -L riscv_ocaml_sim.native coverage.native
 

